core/src/components/segment/segment.tsx

```diff
@@ -431,12 +433,74 @@ export class Segment implements ComponentInterface {
     this.checked = current;
   }
 
+  private getSegmentButton = (selector: 'first' | 'last' | 'next' | 'previous' | 'focused'): HTMLIonSegmentButtonElement | null => {
+    const buttons = this.getButtons().filter(button => !button.disabled);
+    const currIndex = buttons.findIndex(button => button === this.checked || button.classList.contains('ion-focused'));
+    switch (selector) {
+      case 'first':
+        return buttons[0];
+      case 'last':
+        return buttons[buttons.length - 1];
+      case 'next':
+        return buttons[currIndex + 1] || buttons[0];
+      case 'previous':
+        return buttons[currIndex - 1 ] || buttons[buttons.length - 1];
+      case 'focused':
+        return this.el.querySelector('ion-segment-button.ion-focused');
+      default:
+        return null;
+    }
+  }
+
+  private onKeyUp = (ev: KeyboardEvent) => {
+    const isRTL = document.dir === 'rtl';
+    let current;
+    switch (ev.key) {
+      case 'ArrowRight':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('previous') : this.getSegmentButton('next');
+        break;
+      case 'ArrowLeft':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('next') : this.getSegmentButton('previous')
+        break;
+      case 'Home':
+        ev.preventDefault();
+        current = this.getSegmentButton('first');
+        break;
+      case 'End':
+        ev.preventDefault();
+        current = this.getSegmentButton('last');
+        break;
+      case ' ':
+      case 'Enter':
+        ev.preventDefault();
+        current = this.getSegmentButton('focused');
+      default:
+        break;
+    }
+
+    const previous = this.checked || current;
+    if (previous && current) {
+      this.checkButton(previous, current);
+    }
+  }
+
+  private handleFallbackFocus() {
```

Comment:
What is the purpose of this?

Reply:
Each `ion-segment-button` sets their `tabIndex` to `-1` if they are not selected. That way, the `selected` button will always receive focus on tab.

This function fixes an edge case where *none* of the segment buttons are selected, since that is possible if the user doesn't set a `value`. In that case, all of the elements would have `tabIndex={-1}`. If that situation is met, this function makes the first button tab-able.